specs/Standards/Tokens/FungibleTokenMetadata.md

```diff
@@ -0,0 +1,82 @@
+
+### Examples scenario
+
+#### Token provides metadata upon deploy and initialization
+
+Alice deploys a wBTC fungible token contract.
+
+**Assumptions**
+
+- The wBTC token contract is `wbtc`.
+- Alice's account is `alice`.
+- The precision ("decimals" in this metadata standard) on wBTC contract is `10^8`.
+
+**High-level explanation**
+
+Alice issues a transaction to deploy and initialize the fungible token contract, providing arguments to the initialization function that set metadata fields.
+
+**Technical calls**
+
+1. `alice` deploys a contract and calls `wbtc::new({"owner_id": "wbtc", "total_supply": "100000000000000", "version": "w-0.1.0", "name": "Wrapped Bitcoin", "symbol": "WBTC", "reference": "https://example.com/wbtc.json", "reference_hash": "7c879fa7b49901d0ecc6ff5d64d7f673da5e4a5eb52a8d50a214175760d8919a", "decimals": 8})`.
+
+If this deploy and initialization were done using [NEAR CLI](https://docs.near.org/docs/tools/near-cli) the command would be:
+
+    near deploy wbtc --wasmFile res/ft.wasm --initFunction new --initArgs '{"owner_id": "wbtc", "total_supply": "100000000000000", "version": "w-0.1.0", "name": "Wrapped Bitcoin", "symbol": "WBTC", "reference": "https://example.com/wbtc.json", "reference_hash": "7c879fa7b49901d0ecc6ff5d64d7f673da5e4a5eb52a8d50a214175760d8919a", "decimals": 8}'
+
+## Reference-level explanation
+
+A fungible token contract implementing the metadata standard shall contain a field named `ft_metadata`. The field will link to a structure with the properties from the interface below.
+
+**Interface**:
+
+```ts
+type FungibleTokenMetadata {
+    version: string;
+    name: string;
+    symbol: string;
+    reference: string;
+    reference_hash: string;
+    decimals: number;
+}
+```
+
+**Fields**:
+
+- `version` is a string and is open for interpretation. That is to say, conventions will likely arise but are not enforced. It's not necessarily using semantic versioning. There may, for instance, be a benefit in having a version prefix/suffix specificially for wrapped or bridged tokens.
+- `name` is the human-readable name of the token.
+- `symbol` is the abbreviation, like wETH or AMPL.
+- `reference` is a link to a valid JSON file containing various keys offering supplementary details on the token. (For example: "/ipfs/QmdmQXB2mzChmMeKY47C43LxUdg1NDJ5MWcKMKxDu7RgQm", "https://example.com/token.json", etc.)
+- `reference_hash` is the sha256 hash of the JSON file contained in the `reference` field. This is to guard against off-chain tampering.
+- `decimals` is used aid in the frontend showing the proper significant digits of a token. This concept is explained well in this [OpenZeppelin post](https://docs.openzeppelin.com/contracts/3.x/erc20#a-note-on-decimals).
+
+## Drawbacks
+
+- It could be argued that `symbol` and even `name` could belong as key/values in the `reference` JSON object.
```

Comment:
I think given NEAR storage costs it makes sense to inline most of the metadata. Especially since `reference_hash` already disallows having mutable metadata (which would be a benefit of having separate reference), kinda not clear why force clients make another HTTP (or worse IPFS) request.

Reply:
like your typical Rust contract is about 300kb+, metadata likely not gonna be more than single digit kb.

Reply:
hey, a lot has changed since this comment, but `reference` is not required. here's the current metadata. Sadly I don't think we can continue to iterate on this or else I will be decapitated. 😆 

```ts
type FungibleTokenMetadata = {
    spec: string;
    name: string;
    symbol: string;
    icon: string|null;
    reference: string|null;
    reference_hash: string|null;
    decimals: number;
}
```